### ISSUE TITLE
Abandonware procedure

### ISSUE THREAD

**Issue by XAMPPRocky** [rust-lang/wg-governance#22]
The discussion around access policy, had me thinking about how can we handle projects that have been or become abandoned? Over time there will be projects that fall to the way side for one reason or another, and the original authors for whatever reason might not be available or able to announce the deprecation. It's important that these projects are clearly marked as abandoned so that people aren't confused about what project's the Rust teams are and are not maintaining.

### Potential procedures

* **On demand** Given that this would be unlikely to happen often, we could provide a communication channel to let people direct attention to a project's status.
 * **Triage** There could be a regular schedule to look for projects that have languished in the organisation. Since this would be likely need to be done infrequently it could be done on an annual basis.
    * If it's annual it could coincide with All-hands, which could make it easier to have a group consensus on a decision.

### Other questions
* How should should the deprecation of a project be communicated to the community? Some form of announcement should be made to inform, at the same time it doesn't have to made as a blog post.
* Other people or organisations might be interested in taking care of an abandoned project. How do we handle that? Would we transfer ownership?
* Who should be tasked with handling this decision? Though it's unlikely, deprecation of abandonware is potentially sensitive or controversial so it should probably fall to core, or one or more relevant team leads to make the decision.

**Comment by BatmanAoD** [comment]
Do you have a specific definition of "abandoned" in mind, or should that be listed as another open question?

**Comment by Ixrec** [comment]
This seems closely related to the various "namespace" and "crate transfer" proposals that pop up from time to time. The last big discussion in that ballpark which I'm aware of was https://internals.rust-lang.org/t/pre-erfc-crate-name-transfer/9027 that led to https://github.com/rust-lang/rfcs/pull/2614. The big summary comment at the end of all that was https://github.com/rust-lang/rfcs/pull/2614#issuecomment-457396210 from January this year, though it seems like discussion died out shortly after that so the RFC never got pushed to consensus.

**Comment by XAMPPRocky** [comment]
> Do you have a specific definition of "abandoned" in mind, or should that be listed as another open question?

I don't think we should have any really specific criteria of "abandoned", details like "time since last update" or "update frequency" are all things which are context dependent and deciding what the cutoffs for what would or would not make a project "abandoned" would make it feel like the decision was arbitrary. I think abandonware is an "We'll know it when we see it" situation, and thus I want to focus on rather how we can make sure we see it when it's happening, and prevent things falling through the cracks.

As a result, the only criteria I think we consider is whether original author(s) are contactable to make a final decision on the project's deprecation. To me, this is the most flexible, and feels the most fair.

I think there's also space for a "How to deprecate a project" procedure that can used for both abandonware as well as deliberately deprecated projects across teams.

**Comment by XAMPPRocky** [comment]
@Ixrec Thank you for the links! I just read all of it, and from my understanding that relates to rust-lang teams transferring abandoned third party crates. I think we should limit the scope of this procedure to just projects that the rust-lang teams and their related organisations (e.g. `rust-lang-nursery`) manage.

**Comment by Ixrec** [comment]
Ah, I completely missed that that was your intent in the original post. Definitely an important clarification.

**Comment by BatmanAoD** [comment]
>  I think abandonware is an "We'll know it when we see it" situation...

"The ${X} team has declared it abandoned" would be a perfectly suitable definition, but what is X? Is it the governance WG?

I think there's also some important consideration to be had around what "abandoned" means from a user perspective, since that may or may not mean that it shouldn't be used. E.g. if a security bug is discovered in a project that's no longer maintained, I think users would want to know that; but there are other cases in which it doesn't matter too much when a project stops receiving updates (e.g. the original Markdown rendering script).
